ksql-serde/src/test/java/io/confluent/ksql/logging/processing/LoggingDeserializerTest.java

```diff
@@ -0,0 +1,137 @@
+import java.util.function.Function;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LoggingDeserializerTest {
+
+  private static final GenericRow SOME_ROW = new GenericRow(ImmutableList.of("some", "fields"));
+  private static final byte[] SOME_BYTES = "some bytes".getBytes(StandardCharsets.UTF_8);
+  private static final ProcessingLogConfig LOG_CONFIG = new ProcessingLogConfig(ImmutableMap.of(
+      ProcessingLogConfig.INCLUDE_ROWS, true
+  ));
+
+  @Mock
+  private Deserializer<GenericRow> delegate;
+  @Mock
+  private ProcessingLogger processingLogger;
+  @Captor
+  private ArgumentCaptor<Function<ProcessingLogConfig, SchemaAndValue>> errorCaptor;
+
+  private LoggingDeserializer deserializer;
+
+  @Before
+  public void setUp() {
+    deserializer = new LoggingDeserializer(delegate, processingLogger);
+  }
+
+  @Test
+  public void shouldThrowNPEs() {
+    new NullPointerTester().testAllPublicConstructors(LoggingDeserializer.class);
+  }
+
+  @Test
+  public void shouldConfigureDelegate() {
+    // Given:
+    final Map<String, ?> configs = ImmutableMap.of("some", "thing");
+
+    // When:
+    deserializer.configure(configs, true);
+
+    // Then:
+    verify(delegate).configure(configs, true);
+  }
+
+  @Test
+  public void shouldCloseDelegate() {
+    // When:
+    deserializer.close();
+
+    // Then:
+    verify(delegate).close();
+  }
+
+  @Test
+  public void shouldDeserializeWithDelegate() {
+    // Given:
+    when(delegate.deserialize(any(), any())).thenReturn(SOME_ROW);
+
+    // When:
+    deserializer.deserialize("some topic", SOME_BYTES);
+
+    // Then:
+    verify(delegate).deserialize("some topic", SOME_BYTES);
+  }
+
+  @Test(expected = ArithmeticException.class)
+  public void shouldThrowIfDelegateThrows() {
+    // Given:
+    when(delegate.deserialize(any(), any())).thenThrow(new ArithmeticException());
+
+    // When:
+    deserializer.deserialize("t", SOME_BYTES);
+
+    // Then: throws
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void shouldLogOnException() {
+    // Given:
+    when(delegate.deserialize(any(), any()))
+        .thenThrow(new RuntimeException("outer",
+            new RuntimeException("inner", new RuntimeException("cause"))));
+
+    try {
+      // When:
+      deserializer.deserialize("t", SOME_BYTES);
+    } catch (final RuntimeException e) {
+      // Then:
+      verify(processingLogger).error(errorCaptor.capture());
+
+      final SchemaAndValue result = errorCaptor.getValue().apply(LOG_CONFIG);
+
+      assertThat(result, is(SerdeProcessingLogMessageFactory
+          .deserializationErrorMsg(e, Optional.of(SOME_BYTES)).apply(LOG_CONFIG)));
```

Comment:
Use a more specific exception here? This way we ensure we log the error that's thrown by deserialize:

```
final Exception e = new RuntimeException("outer", new RuntimeException("inner", new RuntimeException("cause")));
when(delegate.deserialize(any(), any())).thenThrow(e);
...
} catch (final Exception caught) {
    assertThat(result, is(SerdeProcessingLogMessageFactory
          .deserializationErrorMsg(e, Optional.of(SOME_BYTES)).apply(LOG_CONFIG)));
}
```

Reply:
The test above tests the right type of exception is thrown. No need to test it twice.

Reply:
I don't care about the expected exception from the test. I mean we should be more strict about what we expect to get logged to processing log.